src/compute-client/src/metrics.rs

```diff
@@ -163,6 +169,21 @@ impl ComputeControllerMetrics {
                 var_labels: ["instance_id", "result"],
                 buckets: histogram_seconds_buckets(0.000_500, 32.),
             )),
+            connected_replica_count: metrics_registry.register(metric!(
+                name: "mz_compute_controller_connected_replica_count",
+                help: "The number of replicas successfully connected to the compute controller.",
+                var_labels: ["instance_id"],
+            )),
+            replica_connects_total: metrics_registry.register(metric!(
+                name: "mz_compute_controller_replica_connects_total",
+                help: "The total number of replica (re-)connections made by the compute controller.",
+                var_labels: ["instance_id", "replica_id"],
+            )),
+            replica_connect_wait_time_seconds_total: metrics_registry.register(metric!(
+                name: "mz_compute_controller_replica_connect_wait_time_seconds_total",
+                help: "The total time the compute controller spent waiting for replica (re-)connection.",
+                var_labels: ["instance_id", "replica_id"],
+            )),
```

Comment:
Should this be a histogram?

Reply:
My guess is that between the two metrics "number of connects" and "total wait time" we should have enough information to not require a histogram. At least it will let us differentiate between a replica that takes very long to connect and a replica in a connect-disconnect cycle. Can always add a histogram later if we notice that we are missing insight after all.